File: 30DaysOfCode/Day 26 - Nested Logic.java
import java.io.*;
import java.util.*;

public class Solution {

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int dexpect = in.nextInt();
        int mexpect = in.nextInt();
        int yexpect = in.nextInt();
        int doriginal = in.nextInt();
        int moriginal = in.nextInt();
        int yoriginal = in.nextInt();
        
        int ddiff = (dexpect-doriginal);
        int mdiff = (mexpect-moriginal);
        int ydiff = (yexpect-yoriginal); 
        
        int fine=0;
                
        if (ydiff > 0) {
            fine = 10000;
        }
        else if (mdiff > 0 && ydiff == 0) {
            fine = 500 * mdiff;
        }
        else if (ddiff > 0 && ydiff == 0) {
            fine = 15 * ddiff;
        }
        
        System.out.println(fine);
    }
}

Sample Input:

9 6 2015
6 6 2015

Sample Output:

45

File: 30DaysOfCode - Kotlin/Day_26_Nested_Logic.kt
import java.util.*;
import kotlin.math.roundToInt


fun main(args: Array<String>) {
    val input = Scanner(System.`in`);
    val day1 = input.nextInt()
    val month1 = input.nextInt()
    val year1 = input.nextInt()

    val day2 = input.nextInt()
    val month2 = input.nextInt()
    val year2 = input.nextInt()

    var fine = 0;

    if(year1 > year2) {
        fine = 10000;
    } else  if ((year1 >= year2) && (month1 > month2)){
        fine = 500 * (month1 - month2);
    } else  if ((year1 >= year2) && (month1 >= month2) && (day1 > day2)){
        fine = 15 * (day1 - day2);
    } else {
        fine = 0;
    }

    println(fine)
}
